### ISSUE TITLE
A request: Would you please explain the type of data used

### ISSUE THREAD

**Issue by CleverCleverFish** [larryshaw0079/CoSleep#3]
I encounter a very serious problem when trying your code because I don't know the meanings of  "data-path-v1" and "data-path-v2" clearly.QAQ
I will be very grateful if you could spare your time to answer my question. (｡>∀<｡)

**Comment by larryshaw0079** [comment]
Hi! Thank you for your concern. The two data paths "data-path-v1" and "data-path-v2" are folders containing numpy data files. The format would like be "subj1.npy, subj2.npy ...."

**Comment by CleverCleverFish** [comment]
> 

Thank you for your reply. Could you please give more details of your data processing? 

To be honest, I have encountered many problems to figure out which way of processing data could be adapted to your code perfectly  (っ╥╯﹏╰╥c)

**Comment by CleverCleverFish** [comment]
> Hi! Thank you for your concern. The two data paths "data-path-v1" and "data-path-v2" are folders containing numpy data files. The format would like be "subj1.npy, subj2.npy ...."

Thank you for your reply. Could you please give more details of your data processing?

To be honest, I have encountered many problems to figure out which way of processing data could be adapted to your code perfectly (っ╥╯﹏╰╥c)

**Comment by larryshaw0079** [comment]
Whats your problems. Could you explain them in detail?

**Comment by CleverCleverFish** [comment]
> Whats your problems. Could you explain them in detail?

Firstly, I don't know how to transfer a timing signals into the image data by HHT as you shows. What I can do is only to compute every IMF. So It's difficult to get the images...emm,just for me. I the related code could be shared, you will be highly re
Secondly, I don't know the size of the data you input. I mean the number of  dimensions and the meaning of singal dimension. 
Thirdly, I have run your code, and I also have many problems in this process. I have a strong desire to know what the good curve of the loss or accuaracy trend is like and how many epochs or iterations  have you spent on a entire training process. When running it, I usually find that the pretraining loss don't continue to fall...and it's really at an comparatively early epoch.
Fourthly, both process of finetuning and pretraining have encountered the serious problem of class disequilibrium, which lead to high accuracy with high loss in pretraining. So I  wonder how did you solve the problem.
Fifthly, although it may be a silly question, but I still get confused. Your files are a lot and I can't affirm the use of every file. If you could explain them in detail, you must be highly appreciated.


I am sorry that I have so many problems.QAQ
I really appreciate that you can answer them and I am looking forward to your reply.

**Comment by larryshaw0079** [comment]
1. For the 2d image processing, this part is completed by a matlab code using EMD. The processing is just like getting a Spectrogram from a signal. Spectrogram images are computed from each channel separately. Please refer information from google searching such as "calculating spectrogram using emd and hht" since the preprocessing was finished by another author.
2. The dimensions of every processing steps are described in the comments. Take DEAP for example, the dimensions of raw data (https://github.com/larryshaw0079/CoSleep/blob/master/cosleep/data/dataset.py#LL58C45-L58C45) are commented. Most of datasets are arranged data files into subjects, e.g. each data file represents a subject. And a data file contains several trials while each trail have a time dimension and a channel dimension. Please refer the original dataset website since the dimension arrangement can be slightly different between different datasets. 
3. Before you actually running the real dataset, I suggest you feed a random array into the model and check the intermediate calculation steps.

**Comment by CleverCleverFish** [comment]
> 1. For the 2d image processing, this part is completed by a matlab code using EMD. The processing is just like getting a Spectrogram from a signal. Spectrogram images are computed from each channel separately. Please refer information from google searching such as "calculating spectrogram using emd and hht" since the preprocessing was finished by another author.
> 2. The dimensions of every processing steps are described in the comments. Take DEAP for example, the dimensions of raw data (https://github.com/larryshaw0079/CoSleep/blob/master/cosleep/data/dataset.py#LL58C45-L58C45) are commented. Most of datasets are arranged data files into subjects, e.g. each data file represents a subject. And a data file contains several trials while each trail have a time dimension and a channel dimension. Please refer the original dataset website since the dimension arrangement can be slightly different between different datasets.
> 3. Before you actually running the real dataset, I suggest you feed a random array into the model and check the intermediate calculation steps.

Thank you very much~
